src/components/NewTodo/NewTodo.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from 'react';
+import './NewTodo.scss';
+import users from '../../api/users';
+import { NewTodoTypes } from '../../constants/proptypes';
+
+const newTodoForm = {
+  newTitle: {
+    htmlFor: 'title',
+    label: 'Title:',
+    placeholder: 'ToDo title',
+    default: '',
+    isLimited: titleText => (titleText.length > 20
+      ? 'Maximum length of Title'
+      : ''),
+    isError: titleText => (!titleText ? 'Empty title' : ''),
+  },
+  newUser: {
+    htmlFor: 'user',
+    placeholder: 'Select User',
+    default: 0,
+    isError: userId => (!userId ? 'No user selected' : ''),
+  },
+};
+const { newTitle, newUser } = newTodoForm;
+
+export default class NewTodo extends Component {
+  state = {
+    errorMessage: '',
+    title: newTitle.default,
+    user: newUser.default,
+  };
+
+  onFormSubmit = (e) => {
+    e.preventDefault();
+    const { title, user } = this.state;
+
+    if (!newTitle.isError(title) && !newUser.isError(user)) {
+      this.props.onAdd({
+        userId: Number(user),
+        title,
+      });
+      this.setState({
+        user: newUser.default,
+        title: newTitle.default,
+      });
+    } else {
+      this.setState({
+        errorMessage: `${newTitle.isError(title)} ${newUser.isError(user)}`,
+      });
+    }
+  };
+
+  onFormChange = ({ name, value }) => {
+    this.setState({
+      [name]: value,
+      errorMessage: '',
+    });
+  };
+
+  onTitleChange = (target) => {
+    const { value } = target;
+
+    if (newTitle.isLimited(value)) {
+      this.setState({
+        errorMessage: newTitle.isLimited(value),
+      });
+    } else {
+      this.onFormChange(target);
+    }
+  };
+
+  render() {
+    const {
+      onFormSubmit,
+      onFormChange,
+      onTitleChange,
+      state: {
+        errorMessage,
+        title,
+        user,
+      },
+    } = this;
+
+    return (
+      <form onSubmit={e => onFormSubmit(e)}>
```

Comment:
```suggestion
      <form onSubmit={onFormSubmit}>
```
эффект будет такой же, он автоматически прокинет параметры